### PR TITLE
Reject float literals with two leading signs

### PR DESCRIPTION
**Pull request by dneto0**
E.g. --1 should be rejected.

### CODE REVIEW

File: include/util/hex_float.h [hunk 1]
```diff
@@ -711,13 +711,37 @@ std::ostream& operator<<(std::ostream& os, const HexFloat<T, Traits>& value) {
   return os;
 }
 
+// Returns true if the negate_value is true and the next character on the
```
Comment (AWoloszyn) [inline include/util/hex_float.h]: Returns true if the negate_value parameter is true 
or 
Returns true if negate_value is true.

**Comment by AWoloszyn** [comment]
-1 for comment other than that looks good

Also I expect +1.0p3 will fail parse (I don't think we handle unary + in hex-float), but that is a separate issue.

**Comment by dneto0** [comment]
All feedback addressed.  Amended the previous commit.

File: test/TextToBinary.Constant.cpp [hunk 2]
```diff
@@ -279,6 +279,48 @@ INSTANTIATE_TEST_CASE_P(
...
+  uint32_t width;
+  std::string literal;
+};
+
+using OpConstantInvalidFloatConstant = spvtest::TextToBinaryTestBase<
+    ::testing::TestWithParam<InvalidFloatConstantCase>>;
+
+TEST_P(OpConstantInvalidFloatConstant, Samples) {
+  // Check both kinds of instructions that take literal floats.
+  for (const auto& instruction : {"OpSpecConstant", "OpSpecConstant"}) {
```
Comment (dekimir) [inline test/TextToBinary.Constant.cpp]: Is it me, or are both elements of the initializer list the same??

Comment (dneto0) [inline test/TextToBinary.Constant.cpp]: Whoops.  Yes, first one should be OpConstant.

**Comment by dneto0** [comment]
All feedback addressed.

**Comment by dekimir** [comment]
+2

**Comment by dneto0** [comment]
Squashed and pushed into master via 000cad9cc6b

**Pull request closed without merging**